src/agent-proxy.ts

```diff
@@ -149,7 +154,103 @@ export class AgentProxy {
     });
   }
 
-  handleRequestStream(stream: Stream) {
+  handleRequestStream(stream: Stream, data?: Buffer) {
+    if (data) {
+      const header = JSON.parse(data.toString()) as StreamHeader;
+      const protocol = header.target.split("://")[0];
```

Comment:
what if `header.target` is `undefined`? Is that possible scenario?

Reply:
Only if a client is buggy... but very good point. Will add a bit more protection.